packages/core/admin/server/src/services/homepage.ts

```diff
@@ -92,11 +92,11 @@ const createHomepageService = ({ strapi }: { strapi: Core.Strapi }) => {
     });
   };
 
-  const formatDocuments = (documents: Modules.Documents.AnyDocument[], meta: DocumentMeta) => {
+  const formatDocuments = (documents: Modules.Documents.AnyDocument[], meta: DocumentMeta): RecentDocument[] => {
     return documents.map((document) => {
       return {
         documentId: document.documentId,
-        locale: document.locale,
+        locale: document.locale ?? null,
```

Comment:
To debug this I queried the document in the addStatusToDoucment function:

```
const document = strapi.documents(uid).findOne({documentId: recentDocument.documentId})
```

and then picked the fields one at a time to see what `metadataService.getMetadata`` is expecting. I don't think that thing is typed properly if at all.